R/target_market_share.R

```diff
@@ -53,13 +56,38 @@
 #'     region_isos = region_isos_demo,
 #'     by_company = TRUE
 #'   )
+#'
+#' # Calculate unweighted targets
+#' matched %>%
+#'   target_market_share(
+#'     ald = ald_demo,
+#'     scenario = scenario_demo_2020,
+#'     region_isos = region_isos_demo,
+#'     weight_production = FALSE
+#'   )
 target_market_share <- function(data,
                                 ald,
                                 scenario,
                                 region_isos = r2dii.data::region_isos,
                                 use_credit_limit = FALSE,
-                                by_company = FALSE) {
-  stopifnot(is.data.frame(data))
+                                by_company = FALSE,
+                                weight_production = TRUE) {
+  stopifnot(
+    is.data.frame(data),
+    is.data.frame(ald),
+    is.data.frame(scenario),
+    is.data.frame(region_isos),
+    is.logical(use_credit_limit),
+    is.logical(by_company),
+    is.logical(weight_production)
+  )
+
+  if (by_company & weight_production) {
+    warn(message = "You have selected to view the weighted production at company
```

Comment:
Think if it makes sense to create a new class of warning.

Reply:
Do we need a warning when both are FALSE?
![image](https://user-images.githubusercontent.com/5856545/90644903-ecd9ed00-e1fa-11ea-9483-db9a14cb2098.png)

Reply:
Ah woops, no need for a warning when both are FALSE! You may have heard Klaus refer to this as the "client-view". It's not the default, but it's useful